### PR TITLE
DCOS-8601: Ensure package detail links open in new tab

### PR DESCRIPTION
**Pull request by natmegs**
Ensure links on package detail page rendered from markdown open in new tab.

Closes DCOS-8601.

## Testing

- Go to catalog page
- Click on spark
- Ensure link in Description section opens in new tab
- Back to catalog page
- Click on kubernetes
- Ensure link in pre install section opens in new tab

## Trade-offs

Some links were opening in new tabs already and some were not - the ones that were not were being rendered from markdown. This makes wrapping the individual links in a component difficult since they are not separated from other text. We are currently using a package called `marked` to parse the markdown in the StringUtil `parseMarkdown` method. I've modified the `marked` renderer to add target="_blank" to links. This affects links in the Description, Pre Install notes, and Post Install notes sections.

For a discussion about modifying `marked` links, see [this PR](https://github.com/markedjs/marked/pull/659) from the `marked` project.

The StringUtil method is also used in FrameworkConfiguration. I've made the assumption that we want the behaviour to be consistent in both locations.

### CODE REVIEW

File: src/js/utils/StringUtil.js [hunk 1]
```diff
@@ -191,6 +191,14 @@ const StringUtil = {
       return null;
     }
 
+    const renderer = new marked.Renderer();
```
Comment (DanielMSchmidt) [inline src/js/utils/StringUtil.js]: Would you mind adding some test cases for this? maybe something that could break it like

```md
[foo](http://a.com/<a-mean-uri)
```

Comment (natmegs) [reply src/js/utils/StringUtil.js]: What would be the ideal behaviour in that case?

**Comment by natmegs** [comment]
@DanielMSchmidt Added some tests for StringUtil#parseMarkdown if you don't mind taking a look! (@GeorgiSTodorov too)

File: src/js/utils/StringUtil.js (hunk 1 above)
Comment (tnguyen0) [inline src/js/utils/StringUtil.js]: Nit: I wonder if the changes here and `marked.setOptions` need to done every time or can it be done once on import.

File: tests/pages/catalog/packages/package/PackageTab-cy.js [hunk 2]
```diff
@@ -44,53 +41,57 @@ describe("Package Detail Tab", function() {
         );
     });
 
+    it("contains _blank target for links in description", function() {
+      cy.get(".pod.pod-shorter.flush-top.flush-right.flush-left")
```
Comment (tnguyen0) [inline tests/pages/catalog/packages/package/PackageTab-cy.js]: Nit: Do you think we can use get on something with more meaning? I'm not sure if `contains("Description")` would work.

File: src/js/utils/StringUtil.js [hunk 3]
```diff
@@ -191,6 +191,14 @@ const StringUtil = {
...
+    renderer.link = function() {
+      var out = marked.Renderer.prototype.link.apply(this, arguments);
```
Comment (brandonc) [inline src/js/utils/StringUtil.js]: :neckbeard: could be `const`

**Comment by DanielMSchmidt** [comment]
Leaving this unmerged to give @natmegs a chance to react to @brandonc and @tnguyen0 comment :) Would be fine with merging it like this 🤷‍♂️

**Comment by natmegs** [comment]
@tnguyen0 @brandonc @DanielMSchmidt lots of great feedback on this one, thanks! I've changed the integration test to look for an `h2` containing "Description" to check the description links, and I've added a `markdownRenderer` object to the `StringUtil` to wrap the `marked` functionality. I think it would be great to have this in a separate MarkdownUtil, but I wonder if it would be worth doing that refactor in a separate ticket? Once this one is approved I will add these changes to the backport PRs as well

**Pull request merged**